docs/adr/adr-007-monitoring-and-alerting.md

```diff
@@ -0,0 +1,234 @@
+control of GOV.UK engineers. Finally, by nature of being time based, these
+alerts are also vulnerable to the system being busy - where a sufficiently
+high volume of travel advice publishings guarantees that an alert will
+trigger.
+
+Reviewing these helped us understand the reputation for unactionable alerts and
+reviewing the GOV.UK incidents for Email Alert API helped us to identify
+patterns where incidents reflected ambiguity over whether the system
+was broken or not. We felt there was room for improvement.
+
+[monitor-time-frame]: https://github.com/alphagov/email-alert-api/blob/c70fcd7e5d64393074a3eebd83b3467fd8cc03b3/app/workers/digest_run_worker.rb#L6-L9
+[monitor-workload]: https://github.com/alphagov/govuk-puppet/blob/fac4b5163f10736bb19aa51a8eb7bc276dd2cb40/modules/govuk/manifests/apps/email_alert_api/sidekiq_queue_check.pp#L33-L44
+[email-alert-monitoring]: https://github.com/alphagov/email-alert-monitoring
+[Travel Advice Publisher]: https://github.com/alphagov/travel-advice-publisher
+
+## Decision
+
+Our decision is that the system should raise a critical alert
+[if and only if](https://en.wikipedia.org/wiki/If_and_only_if) the email alert
+system is broken. The majority of these alerts should contact engineers
+out-of-hours.
+
+Unpacking this, we consider the email alert system to be comprised of 3
+applications: Email Alert API,
+[Email Alert Frontend](https://github.com/alphagov/email-alert-frontend) and
+[Email Alert Service](https://github.com/alphagov/email-alert-service); this
+excludes the monitoring of whether GOV.UK publishing is working and whether
+Notify is working as they are independently monitored. The use of
+"if and only if" reflects a condition we're forming that the system will not
+raise critical alerts unless we're highly confident it is broken, whereas
+warnings are appropriate for other situations. We decided that for the email
+alert system to be "broken" it represents a failing that will lead to a user
+noticeable issue unless an engineer intervenes. Since these alerts should
+represent a broken system requiring intervention, it should be suitable to
+contact engineers out-of-hours for them as otherwise users will be
+experiencing issues.
+
+We decided that we will re-approach our alerting with this mindset and look to
+replace alerts that are vulnerable to reflecting a busy status.
+
+## Status
+
+Pending
+
+## Consequences
+
+We will be changing the approaching to monitoring and alerting across the email
+alert system of applications in order to meet this new alerting criteria.
+
+## Monitoring Email Alert API
+
+### Sidekiq workers
+
+We will configure critical alerts, that contact engineers out-of-hours, if
+individual Sidekiq jobs fail multiple times for the following workers:
+
+- [DailyDigestInitiatorWorker][]
+- [DigestEmailGenerationWorker][]
+- [ProcessContentChangeWorker][]
+- [ProcessMessageWorker][]
+- [SendEmailWorker][]
```

Comment:
Sure, that's OK. I was trying to remember it's behaviour regarding rate limiting, which lead me to suggest this. In general, I think we're asserting that none of these jobs should be prone to failure, so we can treat them all equally.

Does that match your understanding?

Reply:
Yup, rate limiting doesn't result in a failure (the worker is [rescheduled](https://github.com/alphagov/email-alert-api/blob/1711885b4df9c80aed5d8d2333e4e007c778badd/app/workers/send_email_worker.rb#L16-L17)). `temporary-failure` is also something we hear about in a call back rather than at sending time.

Yeah, from the doc our decision was to treat these the same. With the external service there is a greater risk of a problem in SendEmailWorker but I recall us concluding the odds of 5 intermittent fails in a row were still absurdly low.